backend/capellacollab/sessions/routes.py

```diff
@@ -542,6 +566,149 @@ def create_database_and_guacamole_session(
     )
 
 
+@project_router.post(
```

Comment:
The new endpoint is good, however, we should merge it with the `read-only` endpoint. They have many lines in common and we should reduce code duplication as best as possible.

As described in https://github.com/DSD-DBS/capella-collab-manager/issues/1004, pre-provisioned sessions shall replace read-only sessions. `pre-provisioned` sessions have the same logic like read-only sessions. The only difference is that no persistent volume is passed.

Reply:
I like to keep them separate endpoints for now. This PR focuses on spawning training sessions. Doing a rewrite of read-only sessions will blow up the scope of this PR.

Reply:
I don't like the two endpoints due to the code duplication, which we try to avoid. The duplication increases future code maintainability and the effort to merge those is not too high. The provision endpoint is very similar to the `read-only` endpoint. The provision endpoint shall take a list of models instead of the project name anyway as decided in #1004. 

Given this, there is only one flag to add: `persistent`, which is a boolean that can be set to true or false. With this flag, we get rid of the code duplication and have one merged endpoint.

As this PR implements https://github.com/DSD-DBS/capella-collab-manager/issues/1004, which explicitly mentions that we want to merge both endpoints, I don't see it out of scope.